app/views/templates/nextReturnSection.scala.html

```diff
@@ -38,7 +38,11 @@ <h2 class="heading-medium" style="margin-top: 0px">@messages("returnObligation.n
 
   <p>
     <a id="vat-returns-link" href="@{appConfig.vatReturnDeadlinesUrl}" data-journey-click="return:view-deadlines:vat-details">
-      @messages("returnObligation.viewReturns")
+        @if(isNonMTDfB) {
```

Comment:
There's a scenario in the story that you've missed:
When we can't retrieve the mandation status at all (and feature switch is on), put the following text in the partial:
```
Sorry, there is a problem with the service. 
Try again later.
```
and don't show the 'Submit return' or 'Return deadlines' link. See the story and google doc for the full requirement

Reply:
Oh, yeah. I opened the sub-task up, and didn't have the over-arching ticket open. My bad.